src/procedures/__tests__/TransferSecurityTokens.ts

```diff
@@ -0,0 +1,178 @@
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have TransferSecurityTokens type', async () => {
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.TransferSecurityTokens);
+    });
+  });
+
+  describe('TransferSecurityTokens', () => {
+    test('should throw if there is no valid security token supplied', async () => {
+      tokenFactoryMock
+        .mock('getSecurityTokenInstanceFromTicker')
+        .withArgs(params.symbol)
+        .throws();
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `There is no Security Token with symbol ${params.symbol}`,
+        })
+      );
+    });
+  });
+
+  test('should add a transaction to the queue to execute a transfer security token using a different sender address', async () => {
+    target = new TransferSecurityTokens(
+      { ...params, from: '0x1FB52cef867d95E69d398Fe9F6486fAF92C7ED7F' },
+      contextMock.getMockInstance()
+    );
+    contextMock.set(
+      'currentWallet',
+      new Wallet({ address: () => Promise.resolve('0x0e6b236a504fce78527497e46dc90c0a6fdc9495') })
+    );
+
+    securityTokenMock.mock(
+      'canTransferFrom',
+      Promise.resolve({
+        statusCode: TransferStatusCode.TransferSuccess,
+      })
+    );
+
+    const addTransactionSpy = spy(target, 'addTransaction');
+    securityTokenMock.mock('transferFromWithData', Promise.resolve('TransferFromWithData'));
+
+    await target.prepareTransactions();
+
+    expect(
+      addTransactionSpy
+        .getCall(0)
+        .calledWith(securityTokenMock.getMockInstance().transferFromWithData)
+    ).toEqual(true);
+    expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(
+      PolyTransactionTag.TransferSecurityTokens
+    );
+    expect(addTransactionSpy.callCount).toEqual(1);
+  });
+
+  test('should throw error if canTransferFrom method returns status code different than success', async () => {
+    const from = '0x1FB52cef867d95E69d398Fe9F6486fAF92C7ED7F';
+    const reasonCode = '0x50';
+
+    target = new TransferSecurityTokens({ ...params, from }, contextMock.getMockInstance());
+    contextMock.set('currentWallet', new Wallet({ address: () => Promise.resolve(from) }));
+
+    securityTokenMock.mock(
+      'canTransfer',
+      Promise.resolve({
+        statusCode: TransferStatusCode.TransferFailure,
+        reasonCode,
+      })
+    );
+
+    await expect(target.prepareTransactions()).rejects.toThrowError(
+      new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `[${TransferStatusCode.TransferFailure}] ${from} is not allowed to transfer ${
+          params.symbol
+        } to ${params.to}. Possible reason: ${reasonCode}`,
+      })
+    );
+  });
+
+  test('should successfully resolve controller transfer', async () => {
```

Comment:
Still incorrect. This is not a controller transfer

Reply:
wait... this is not my change

Reply:
`should successfully refresh the corresponding balance of each shareholder involved`